metagraph/core/planning.py

```diff
@@ -13,25 +13,33 @@ def __init__(self, src_type):
         self.src_type = src_type
         self.translators = []
         self.dst_types = []
+        self.unsatisfiable = False
 
     def __len__(self):
+        if self.unsatisfiable:
+            raise ValueError(
+                "No translation path found for {src_type.__name__} -> {self.dst_types[-1].__name__}"
+            )
         return len(self.translators)
 
     def __iter__(self):
+        if self.unsatisfiable:
+            raise ValueError(
+                "No translation path found for {src_type.__name__} -> {self.dst_types[-1].__name__}"
+            )
         return iter(self.translators)
 
     def __repr__(self):
-        if self.src_type is None or not self.dst_types:
-            return f"{self.__class__.__name__}"
-        return f"{self.__class__.__name__}({self.src_type.__name__} -> {self.dst_types[-1].__name__})"
+        s = []
+        s.append("[Multi-step Translation]")
```

Comment:
You can't always start with "Multi-step Translation". That only applies when `len(self) > 1`.

The display should look like one of these:
```
[Multi-step Translation]
(start)  Foo
           -> Bar
 (end)        -> Baz
```

```
[Direct Translation]
Foo -> Bar
```

```
[Unsatisfiable Translation]
Foo -> Baz
```

Reply:
That makes sense. 

https://github.com/metagraph-dev/metagraph/pull/110/commits/1da1b50f2bc21e152d58a97f476f732d2ffbb6a8 intends to address this problem. 

Here are the possible outcomes:

```
[Unsatisfiable Translation]
Translation to DST_TYPE unsatisfiable"
```
```
[Null Translation]
No translation required
```
```
[Multi-step Translation]
(start)  SRC_TYPE
           -> INTERMEDIATE_TYPE
(end)        -> DST_TYPE
```
```
[Direct Translation]
SRC_TYPE -> DST_TYPE
```